File: python/maxSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
    	s_idx = e_idx = 0
    	numSum = nums[0]
    	for i in range(1, len(nums)):
    		if nums[i] > numSum + nums[i]:
    			numSum = nums[i]
    			s_idx = i
    			e_idx = i
    		else:
    			e_idx += 1
    			

File: cpp/maxSubArray.cpp
#include <iostream>
#include <cstdio>
#include <fstream>
#include <algorithm>
#include <cmath>
#include <deque>
#include <vector>
#include <queue>
#include <string>
#include <cstring>
#include <unordered_map>
#include <stack>
#include <set> 
using namespace std;
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> dp(n, 0);
        dp[0] = nums[0];
        int ans = dp[0];
        for (int i = 1; i < n; i++) {
            dp[i] = max(nums[i], nums[i] + dp[i - 1]);
            ans = max(ans, dp[i]);
        }
        return ans;
    }
};